bundles/worlds/worlds_service.go

```diff
@@ -323,81 +321,106 @@ func (ws *Service) CreateWorldReport(tx *gorm.DB, owner, worldName, reason strin
 }
 
 // RemoveWorldLike removes a worldLike.
-// Returns the removed worldLike, the current count of likes, or an gz.errMsg.
-func (ws *Service) RemoveWorldLike(tx *gorm.DB, owner, worldName string,
-	user *users.User) (*WorldLike, int, *gz.ErrMsg) {
+// Returns the removed worldLike or a gz.errMsg.
+func (ws *Service) RemoveWorldLike(tx *gorm.DB, owner, worldName string, user *users.User) (*WorldLike, *gz.ErrMsg) {
 
 	if user == nil {
-		return nil, 0, gz.NewErrorMessage(gz.ErrorAuthNoUser)
+		return nil, gz.NewErrorMessage(gz.ErrorAuthNoUser)
 	}
 
 	world, em := ws.GetWorld(tx, owner, worldName, user)
 	if em != nil {
-		return nil, 0, em
+		return nil, em
 	}
 
 	// Unlike the world.
 	var worldLike WorldLike
 	if err := tx.Where("user_id = ? AND world_id = ?", &user.ID, &world.ID).Delete(&worldLike).Error; err != nil {
-		return nil, 0, gz.NewErrorMessageWithBase(gz.ErrorDbDelete, err)
+		return nil, gz.NewErrorMessageWithBase(gz.ErrorDbDelete, err)
 	}
 	// Update the number of likes of the world.
-	count, errorMsg := ws.updateLikeCounter(tx, world)
+	errorMsg := ws.decreaseLikeCounter(tx, world)
 	if errorMsg != nil {
-		return nil, 0, errorMsg
+		return nil, errorMsg
 	}
-	return &worldLike, count, nil
+	return &worldLike, nil
 }
 
-// updateLikeCounter counts the number of likes and updates the world accordingly.
-func (ws *Service) updateLikeCounter(tx *gorm.DB, world *World) (int, *gz.ErrMsg) {
-	var counter int
+// applyExpression updates a world using SQL expression that can perform operations on referred values.
+func (ws *Service) applyExpression(tx *gorm.DB, world *World, field string, expr *gorm.SqlExpr) *gz.ErrMsg {
+	if err := tx.Model(world).Update(field, expr).Error; err != nil {
+		return gz.NewErrorMessageWithBase(gz.ErrorDbSave, err)
+	}
+	return nil
+}
```

Comment:
This method receives a `world`, we can probably extract it as a function and receive a `commonres.Resource`, making this function generic for both models and worlds.

Reply:
This comment also applies to the `applyExpression` function in the models service.

Reply:
Right now the Model and World services are pretty much duplicates of each other. I tried to maintain the style in which things were done. I would like to do a more extensive refactor and merge both services, but in a future effort as this PR addresses performance issues that are affecting us today.

Reply:
I'd argue that we can incrementally refactor the current codebase by addressing these issues as they come by while changing other parts of the code. I understand if we don't want to do it now.